src/platform/lumin-runtime/utilities/resource-download.js

```diff
@@ -0,0 +1,74 @@
+// The purpose of the resource downloader is to download image or model from remote location (url)
+
+import { ui, INVALID_RESOURCE_ID } from 'lumin';
+
+import executor from './executor.js';
+
+import { logError } from '../../../util/logger.js';
+import saveResource from '../../../util/download.js';
+
+const DOWNLOAD_COLOR_MASK = [0.1, 0.1, 0.1, 0.1];
+const DEFAULT_COLOR_MASK = [1, 1, 1, 1];
+
+function createSpinner (prism) {
+    const spinner = executor.createNode(ui.UiLoadingSpinner, 'Create', prism, ui.LoadingSpinnerType.k2dSpriteAnimation);
+    const [w, h] = executor.callNativeFunction(spinner, 'getSize');
+    const [x, y, z] = executor.callNativeFunction(spinner, 'getLocalPosition');
+
+    executor.callNativeAction(spinner, 'setLocalPosition', [x - (w / 2), y - (h / 2), z]);
+    return spinner;
+}
+
+function addMaskAndSpinner (element, prism) {
+    // Set color mask
+    executor.callNativeAction(element, 'setColor', DOWNLOAD_COLOR_MASK);
+
+    // Add downloading spinner
+    const spinner = createSpinner(prism);
+    executor.callNativeAction(element, 'addChild', spinner);
+
+    return spinner;
+}
+
+function removeMaskAndSpinner (element, prism, spinner, properties) {
+    // Remove color mask
+    const color = properties.color ? properties.color : DEFAULT_COLOR_MASK
```

Comment:
Please make sure that the color is converted to [r,g,b,a] format. If 'properties.color' = `'red'`, I get exception and the spinner does not disappears.

Reply:
Yes! Thank you @pleszkiewicz !